runtime/runtime-params-estimator/src/rocksdb.rs

```diff
@@ -0,0 +1,303 @@
+    let mut keys: Vec<usize> =
+        iter::repeat_with(|| prng.gen()).take(db_config.setup_insertions).collect();
+    if db_config.sequential_keys {
+        keys.sort();
+    } else {
+        // give it another shuffle to make lookup order different from insertion order
+        keys.shuffle(&mut prng);
+    }
+
+    let gas_counter = GasCost::measure(config.metric);
+
+    for i in 0..db_config.op_count {
+        let key = keys[i as usize % keys.len()];
+        db.get(&key.to_string()).unwrap();
+    }
+
+    let cost = gas_counter.elapsed();
+
+    if db_config.debug_rocksdb {
+        println!("# Cost: {:?}", cost);
+        print_levels_info(&db);
+    }
+
+    drop(db);
+    tmp_dir.close().expect("Could not clean up temp DB");
+
+    if db_config.input_data_path.is_none() {
+        backup_input_data(&data);
+    }
+
+    cost
+}
+
+/// Sequentially insert a number of generated key-value pairs and flushes
+///
+/// Keys are {"1", "2", ... } starting at `key_offset`
+/// Values are different slices taken from `input_data`
+fn sequential_inserts(
+    inserts: usize,
+    value_size: usize,
+    input_data: &[u8],
+    key_offset: usize,
+    db: &DB,
+    force_compaction: bool,
+    force_flush: bool,
+) {
+    for i in 0..inserts {
+        let key = (key_offset + i).to_string();
+        let start = (i * value_size) % (input_data.len() - value_size);
+        let value = &input_data[start..(start + value_size)];
+        db.put(&key, value).expect("Put failed");
+    }
+    if force_flush {
+        db.flush().expect("Flush failed");
+    }
+    if force_compaction {
+        db.compact_range::<&[u8], &[u8]>(None, None);
+    }
+}
+
+/// Insert a number of generated key-value pairs and flushes
+///
+/// Keys are pseudo-random and deterministic based on the seed
+/// Values are different slices taken from `input_data`
+fn prandom_inserts(
+    inserts: usize,
+    value_size: usize,
+    input_data: &[u8],
+    key_seed: u64,
+    db: &DB,
+    force_compaction: bool,
+    force_flush: bool,
+) {
+    let mut prng: XorShiftRng = rand::SeedableRng::seed_from_u64(key_seed);
+    for i in 0..inserts {
+        let key = prng.gen::<u64>().to_string();
+        let start = (i * value_size) % (input_data.len() - value_size);
+        let value = &input_data[start..(start + value_size)];
+        db.put(&key, value).expect("Put failed");
+    }
+    if force_flush {
+        db.flush().expect("Flush failed");
+    }
+    if force_compaction {
+        db.compact_range::<&[u8], &[u8]>(None, None);
+    }
+}
+
+fn input_data(db_config: &RocksDBTestConfig, data_size: usize) -> Vec<u8> {
+    let mut data = vec![0u8; data_size];
+    if let Some(path) = &db_config.input_data_path {
+        let mut input = std::fs::File::open(path).unwrap();
+        input.read_exact(&mut data).unwrap();
+        assert_eq!(input.seek(SeekFrom::End(0)).unwrap(), 0, "Provided input file has wrong size");
```

Comment:
at this point, `std::fs::read` might be eashier.

Reply:
Ah okay, I see what you mean. Yes that should be easier